lib/Cake/View/JsonView.php

```diff
@@ -145,10 +149,22 @@ protected function _serialize($serialize) {
 		}
 
 		if (version_compare(PHP_VERSION, '5.4.0', '>=') && Configure::read('debug')) {
-			return json_encode($data, JSON_PRETTY_PRINT);
+			$json = @json_encode($data, JSON_PRETTY_PRINT);
+		} else {
+			$json = @json_encode($data);
 		}
 
-		return json_encode($data);
+		if (function_exists('json_last_error') && json_last_error() !== JSON_ERROR_NONE) {
```

Comment:
You can backport the [json_last_error()](https://github.com/cakephp/cakephp/blob/3.0/src/basics.php#L132) added to basics.php in 3.0.